lib/ansible/module_utils/facts/virtual/linux.py

```diff
@@ -139,19 +139,32 @@ def get_virtual_facts(self):
             virtual_facts['virtualization_role'] = 'guest'
 
         product_name = get_file_content('/sys/devices/virtual/dmi/id/product_name')
+        sys_vendor = get_file_content('/sys/devices/virtual/dmi/id/sys_vendor')
 
         if product_name in ('KVM', 'KVM Server', 'Bochs', 'AHV'):
             guest_tech.add('kvm')
             if not found_virt:
                 virtual_facts['virtualization_type'] = 'kvm'
                 found_virt = True
 
-        if product_name == 'RHEV Hypervisor':
-            guest_tech.add('RHEV')
+        if sys_vendor == 'oVirt':
+            guest_tech.add('oVirt')
             if not found_virt:
-                virtual_facts['virtualization_type'] = 'RHEV'
+                virtual_facts['virtualization_type'] = 'oVirt'
                 found_virt = True
 
+        if sys_vendor == 'Red Hat':
+            if product_name == 'RHEV Hypervisor':
+                guest_tech.add('RHEV')
+                if not found_virt:
+                    virtual_facts['virtualization_type'] = 'RHEV'
+                    found_virt = True
+            elif product_name == 'RHEL':
```

Comment:
IIUC this is still wrong, it will return "RHV" on plain RHEL virtualization. As I commented earlier you need to look at Family (/sys/devices/virtual/dmi/id/product_family) for 4.4+

Reply:
yeah, and then we can check the family first (and return `RHV` if it's RHV) and only then the product name - I see that also with latest versions, VDSM reports `RHEV Hypervisor` for RHVH

Reply:
so something like this?
```python
if sys_vendor == 'Red Hat':
    if product_name == 'RHEV Hypervisor' and product_family == 'Red Hat Enterprise Linux' :
        guest_tech.add('RHEV')
        ...
    elif product_family == 'RHV' or (product_family == 'Red Hat Enterprise Linux' and product_name == 'RHEL'):
        guest_tech.add('RHV')
        ...
```

Reply:
Or:

```python
if sys_vendor == 'Red Hat':
    if product_family == 'RHV':
        guest_tech.add('RHV')
        ...
    elif product_family == 'RHEV Hypervisor':
        guest_tech.add('RHEV')
        ...
    # and then maybe:
    else:
        ovirt_guest_agent_exists = ...
        if ovirt_guest_agent_exists:
            guest_tech.add('RHEV')
             ...
```
In the long run this would always return RHV and in the meantime we'll return `RHEV` also for 3.6<RHV<4.4.6 but that's not that bad I think if we expect clients to keep processing `RHEV` as an output - I doubt someone really cares about the brand. Maybe the version can be important but I don't see it's returned elsewhere

Reply:
Not sure how it is returned in this version of ansible but in 2.9 where we will backport this logic so it can go to customers it is like this.
https://github.com/ansible/ansible/blob/5a9f4f7bb395f4738d682c87e659267cc3929d76/lib/ansible/module_utils/facts/virtual/linux.py#L92-L94
And users use the value not sure how/why but we have opened bz and ticket on it https://bugzilla.redhat.com/show_bug.cgi?id=1904085

How/where I can check the guest agent?

Reply:
Ah, by 'brand' I meant RHEV vs. RHV. So as from 4.4.6, assuming that's where we'll change the sysinfo settings to always include family=RHV, we'll return `RHV` and for lower versions, we'll return `RHEV` for RHVH as in the snippet.

@nyoxi , can you please suggest how to check for ovirt-guest-agent existence in RHV?

Reply:
>  So as from 4.4.6

it's really just a bug in CL 4.5 until the fix in 4.4.5. We have family RHV everywhere since 4.4 and for 4.4 CL it works fine. Both RHEL and RHVH.

Reply:
> it's really just a bug in CL 4.5 until the fix in 4.4.5. We have family RHV everywhere since 4.4 and for 4.4 CL it works fine. Both RHEL and RHVH.

We're ok with CL 4.5 - the SKU is not there but the family is
The problem is different - it's with 4.3 clusters in RHV 4.4 which is still pretty common it seems

Reply:
Posted https://gerrit.ovirt.org/#/c/ovirt-engine/+/113781/

Reply:
> @nyoxi , can you please suggest how to check for ovirt-guest-agent existence in RHV?

That depends on what exactly you want to test here. You can:

 - look if ovirt-guest-agent-common RPM  (or ovirt-guest-agent RPM on EL6 and older) is installed
 - look if there is virtio channel for the agent
 - look if ovirt-guest-agent.service is up and running

But I agree with what Michal said bellow -- it is better to avoid guest agent related check unless absolutely necessary.

Reply:
> > @nyoxi , can you please suggest how to check for ovirt-guest-agent existence in RHV?
> 
> That depends on what exactly you want to test here. You can:
> 
> * look if ovirt-guest-agent-common RPM  (or ovirt-guest-agent RPM on EL6 and older) is installed
> * look if there is virtio channel for the agent

Yeah, that would be the best check it seems

> * look if ovirt-guest-agent.service is up and running
> 
> But I agree with what Michal said bellow -- it is better to avoid guest agent related check unless absolutely necessary.

So it depends on what we see as "absolutely necessary" - this PR is about identifying VMs as RHV VMS. Without checking ovirt-guest-agent we can't really tell whether VMs that run on plain RHEL are part of RHV or not - if we're ok with not identifying them as RHV VMs, that's one things. But if we do, I guess we'll need to check that or something else in their metadata that is unique to RHV..

Reply:
yes, virtio socket is trivial to check for and it's indeed a great idea for <=4.3 at least...it can be just that one thing to look at.

Reply:
Nobody is checking sockets here, can't we just fix product family set by RHV Manager regardless of cluster version used?